build/lex.go

```diff
@@ -571,13 +632,19 @@ func (in *input) skipPython(p []byte) []byte {
 				rest = p[i:]
 			}
 
-			if !isBlankOrComment(p[i:]) {
-				if !hasPythonContinuation(p[i:]) && c != ' ' && c != '\t' {
-					// Yes, stop here.
-					break
+			if tables.FormatBzlFiles {
+				// In the bzl files mode we only care about the end of the statement, we've found it.
+				break
```

Comment:
by my read, this breaks the for loop just to 'return rest'  -- (and also check for quote).
Instead just 'return rest' here, and after the declaration of quote, you can add a 
defer func() {
  if quote != 0 {
 		in.Error("EOF scanning Python quoted string")
 	}
}()